### PR TITLE
Allow ai to understand if its handcuffed.

### LINKED ISSUE #27460: Neutral mobs cannot be pulled -- they escape pulling when unangered.
## Description

Even when not angry, neutral mobs refuse to be pulled.

**Reproduction**

Try to pull Renault, even as aghost, the little bastard runs away.

**Screenshots**

**Additional context**

Not sure if because of #26868  or just happenstance.

### LINKED ISSUE #28006: Cant drag monkeys in makeshift cuffs
## Description
Title says it all pretty much when attempting to drag a monkey in makeshift cuffs it just drops of the hand indication every split second.

**Reproduction**
- Cuff a monkey in makeshift cuffs.
- Try and drag em.

**Screenshots**
image

**Additional context**
As ghost or player/humanoid.

### PR DESCRIPTION
**Pull request by IProduceWidgets**
related at #30397
maybe useful at #30066
partially fixes #27460
partially fixes #28006

## About the PR

Allows Ai to understand if its handcuffed by checking this precondition.

@osjarw useful?

## Why / Balance

Can be used to avoid ai constantly trying to do stuff it cant do because of handcuffs, when it should instead do something else, such as run away.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

no cl

### CODE REVIEW

**Comment by IProduceWidgets** [comment]
dufuq are these test fails

**Comment by IProduceWidgets** [comment]
What are these flake ass tests even...

File: Content.Shared/Cuffs/SharedCuffableSystem.cs [hunk 1]
```diff
@@ -530,6 +530,23 @@ public bool TryCuffing(EntityUid user, EntityUid target, EntityUid handcuff, Han
...
+
+            if (!TryComp<HandsComponent>(target, out var hands))
+                return false;
+
+            if (cuffable.CuffedHandCount <= 0)
+                return false;
+
+            if (requireFullyCuffed == true && hands.Count > cuffable.CuffedHandCount)
+                return false;
+
```
Comment (metalgearsloth) [inline Content.Shared/Cuffs/SharedCuffableSystem.cs:546]:
```suggestion

            if (requireFullyCuffed && hands.Count > cuffable.CuffedHandCount)
                return false;

```

File: Content.Shared/Cuffs/SharedCuffableSystem.cs [hunk 2]
```diff
@@ -530,6 +530,23 @@ public bool TryCuffing(EntityUid user, EntityUid target, EntityUid handcuff, Han
             return true;
         }
 
+        public bool IsCuffed(EntityUid target, bool requireFullyCuffed = true, CuffableComponent? cuffable = null)
+        {
```
Comment (metalgearsloth) [inline Content.Shared/Cuffs/SharedCuffableSystem.cs:534]: Doc summary please, also Entity<CuffableComponent>

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**